test/model/recent_senders_test.dart

```diff
@@ -0,0 +1,201 @@
+      final streamMessage = eg.streamMessage(id: 100, sender: eg.user(userId: 10),
+        stream: eg.stream(streamId: 1), topic: 'topic');
+
+      model.handleMessage(dmMessage);
+      model.handleMessage(streamMessage);
+      checkMatchesMessages([streamMessage], model);
+    });
+
+    test('add a batch of messages', () {
+      final model = RecentSenders();
+      final stream1 = eg.stream(streamId: 1);
+      final stream2 = eg.stream(streamId: 2);
+      final user1 = eg.user(userId: 1);
+      final user2 = eg.user(userId: 2);
+
+      final messages = [
+        eg.streamMessage(stream: stream1, topic: 'a', sender: user1, id: 300),
+        eg.streamMessage(stream: stream1, topic: 'b', sender: user1, id: 100),
+        eg.streamMessage(stream: stream1, topic: 'a', sender: user2, id: 200),
+        eg.streamMessage(stream: stream1, topic: 'b', sender: user2, id: 400),
+        eg.streamMessage(stream: stream2, topic: 'c', sender: user2, id: 500),
+      ];
+
+      model.handleMessages(messages);
+      checkMatchesMessages(messages, model);
+    });
+
+    test('delete messages', () {
+      final model = RecentSenders();
+      final stream1 = eg.stream(streamId: 1);
+      final user1 = eg.user(userId: 1);
+      final user2 = eg.user(userId: 2);
+
+      final messages = [
+        eg.streamMessage(stream: stream1, topic: 'a', sender: user1, id: 300),
+        eg.streamMessage(stream: stream1, topic: 'b', sender: user1, id: 100),
+        eg.streamMessage(stream: stream1, topic: 'a', sender: user2, id: 200),
+      ];
+
+      model.handleMessages(messages);
+      checkMatchesMessages(messages, model);
+
+      model.handleDeleteMessageEvent(
+        DeleteMessageEvent(id: 1, messageIds: [300, 200],
+          messageType: MessageType.stream, streamId: stream1.streamId, topic: 'a'),
+        Map.fromEntries(messages.map((msg) => MapEntry(msg.id, msg))));
+
+      checkMatchesMessages(
+        [eg.streamMessage(stream: stream1, topic: 'b', sender: user1, id: 100)],
+        model);
+    });
+
+    test('latestMessageIdOfSenderInStream', () {
+      final model = RecentSenders();
+      final stream1 = eg.stream(streamId: 1);
+      final stream2 = eg.stream(streamId: 2);
+      final user1 = eg.user(userId: 1);
+      final user2 = eg.user(userId: 2);
+
+      final messages = [
+        eg.streamMessage(stream: stream1, sender: user1, id: 100),
+        eg.streamMessage(stream: stream1, sender: user1, id: 300),
+        eg.streamMessage(stream: stream1, sender: user1, id: 200),
+      ];
+
+      model.handleMessages(messages);
+
+      check(model)
+        ..latestMessageIdOfSenderInStream(
+            streamId: stream1.streamId, senderId: user1.userId).equals(300)
+        ..latestMessageIdOfSenderInStream(
+            streamId: stream1.streamId, senderId: user2.userId).equals(null)
+        ..latestMessageIdOfSenderInStream(
+            streamId: stream2.streamId, senderId: user1.userId).equals(null);
```

Comment:
Let's write these as three separate test cases (three `test` calls). That way they each get their own name that indicates the key point they're testing.

The setup can be simplified for the latter two cases, just adding a single message.

Reply:
Alternatively the latter cases could get a short one-line comment each, saying what they're testing: "unknown sender", "unknown stream". That may be better, particularly when I look at the similar `…InTopic` case below.